src/prefs.ts

```diff
@@ -579,6 +580,132 @@ export default class TilingShellExtensionPreferences extends ExtensionPreference
             keybindingsDialogGroup.add(row);
         });
 
+        // Import/export/reset section
+        const importExportGroup = new Adw.PreferencesGroup({
+            title: 'Import/Export/Reset',
+            description: `Import/export/reset the settings of ${this.NAME}`,
+        });
+        prefsPage.add(importExportGroup);
+
+        const exportSettingsBtn = this._buildButtonRow(
+            'Export settings',
+            'Export settings',
+            'Export settings to a file',
+            () => {
+                const fc = new Gtk.FileChooserDialog({
+                    title: 'Export settings',
+                    select_multiple: false,
+                    action: Gtk.FileChooserAction.SAVE,
+                    transient_for: window,
+                });
+                fc.set_current_folder(
+                    Gio.File.new_for_path(GLib.get_home_dir()),
+                );
+                fc.add_button('Cancel', Gtk.ResponseType.CANCEL);
+                fc.add_button('Save', Gtk.ResponseType.OK);
+                fc.connect(
+                    'response',
+                    (_source: Gtk.FileChooserDialog, response_id: number) => {
+                        try {
+                            if (response_id === Gtk.ResponseType.OK) {
+                                const file = _source.get_file();
+                                if (!file) throw new Error('no file selected');
+
+                                debug(
+                                    `Create file with path ${file.get_path()}`,
+                                );
+                                const settingsExport = new SettingsExport();
```

Comment:
On line 423 you have a **gioSetting** constant which is the `Gio.Setting` object you could use to reset settings whenever it is needed